tensorflow/lite/kernels/ceil.cc

```diff
@@ -36,25 +36,114 @@ TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
+      // zero point should be zero.
+      TF_LITE_ENSURE_EQ(context, input->params.zero_point, 0);
+      TF_LITE_ENSURE_EQ(context, output->params.zero_point, 0);
+    }
+  }
+
   output->type = input->type;
   TfLiteIntArray* output_size = TfLiteIntArrayCopy(input->dims);
   return context->ResizeTensor(context, output, output_size);
 }
 
+template <typename T>
+TfLiteStatus EvalCeil(const TfLiteTensor* input, TfLiteTensor* output) {
+  optimized_ops::Ceil<T>(GetTensorShape(input), GetTensorData<T>(input),
+                         GetTensorShape(output), GetTensorData<T>(output));
+  return TfLiteStatus::kTfLiteOk;
+}
+
+template <typename T>
+TfLiteStatus EvalCeilQuantized(const TfLiteTensor* input,
+                               TfLiteTensor* output) {
+  const int32_t input_zero_point = input->params.zero_point;
+  const float input_scale = input->params.scale;
+  const int32_t output_zero_point = output->params.zero_point;
+  const float output_scale = output->params.scale;
+
+  const int num_elements = NumElements(input);
+  const T* input_data = GetTensorData<T>(input);
+  T* output_data = GetTensorData<T>(output);
+
+  for (int i = 0; i < num_elements; ++i) {
+    const float input_value = (input_data[i] - input_zero_point) * input_scale;
+    const float ceiled_value = std::ceil(input_value);
+    int32_t quantized_value =
+        static_cast<int32_t>(std::round(ceiled_value / output_scale)) +
+        output_zero_point;
+    // Clamp the value to fit in the range of the output type
+    quantized_value =
+        std::min(std::max(quantized_value,
+                          static_cast<int32_t>(std::numeric_limits<T>::min())),
+                 static_cast<int32_t>(std::numeric_limits<T>::max()));
+
+    output_data[i] = static_cast<T>(quantized_value);
+  }
+  return TfLiteStatus::kTfLiteOk;
+}
+
 TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
   const TfLiteTensor* input;
   TF_LITE_ENSURE_OK(context, GetInputSafe(context, node, kInputTensor, &input));
   TfLiteTensor* output;
   TF_LITE_ENSURE_OK(context,
                     GetOutputSafe(context, node, kOutputTensor, &output));
-  if (input->type != kTfLiteFloat32) {
-    TF_LITE_UNSUPPORTED_TYPE(context, input->type, "Ceil");
-  }
-
-  optimized_ops::Ceil(GetTensorShape(input), GetTensorData<float>(input),
-                      GetTensorShape(output), GetTensorData<float>(output));
 
+  switch (input->type) {
+    case kTfLiteFloat32:
+      TF_LITE_ENSURE_OK(context, EvalCeil<float>(input, output));
+      break;
+    case kTfLiteFloat16:
+      TF_LITE_ENSURE_OK(context, EvalCeil<Eigen::half>(input, output));
+      break;
+    case kTfLiteBFloat16:
+      TF_LITE_ENSURE_OK(context, EvalCeil<Eigen::bfloat16>(input, output));
+      break;
+    case kTfLiteInt32:
+      TF_LITE_ENSURE_OK(context, EvalCeil<int32_t>(input, output));
+      break;
+    case kTfLiteInt8:
+      TF_LITE_ENSURE_OK(context, EvalCeilQuantized<int8_t>(input, output));
```

Comment:
Can you clarify if unquantized int8 is supported? If not, can you please help enable int8 support?